test/utils/gtclang-tester/gtclang_tester/parser.py

```diff
@@ -141,6 +142,36 @@ def __parse_line(self, line, linenumber):
             report_info("Parsed EXPECTED_FILE as: \"%s\"" % expected_file)
             self.__test.add_expected_file_command(expected_file, self.__dir, self.__file,
                                                   linenumber)
+    
+
+    def __parse_line_with_number(self, line, linenumber):
```

Comment:
I don't doubt it works but it's not very [pythonic](https://stackoverflow.com/questions/25011078/what-does-pythonic-mean) ;)
How about something like
```
import re

def __parse_line_with_number(line, linenumber):
    """ Parse a line with optional expresion evaluation """
    pattern = re.search('^(%line(([+-]{1})([0-9]*))?%){1}.*$', line)
    if not pattern:
        return line

    expr = pattern.group(2)
    op = pattern.group(3)
    value = pattern.group(4)
   
    if expr:
        replacement = str(linenumber + int(value) if op == '+' else linenumber - int(value))
    else:
        replacement = str(linenumber)

    return line.replace(pattern.group(1), replacement)
```
I guess a regex is pretty handy in this scenario. The regex itself is pretty simple and just uses 5 groups to match the relevant sub patterns. For example: `%line+5%` will produce a match where:
  - `group(0)` matches the entire line (not very useful in this case)
  - `group(1)` matches `%line+5%` (hence everything we need to replace)
  - `group(2)` matches `+5` (hence indicates if we need to evaluate the expression, in case there is no expression this would be `None`)
 - `group(3)` matches `+`
 - `group(4)` matches `5`